### PR TITLE
feat(tasks): render tasks into Makefile

### PR DESCRIPTION
**Pull request by Chriscbr**
Adds the capability to synthesize tasks into a Makefile format. I believe this is a prerequisite for implementing a solution of #596.

Just experimenting at the moment, this PR isn't close to being ready yet.

- [ ] environment variables not working yet
- [ ] ensure output is deterministic (no direct paths to node etc.)
- [ ] research better ways to handle recursive tasks in make?
- [ ] refactor task code to minimize duplicate code
- [ ] have the default `make` task (when no target is specified) be projen's "default" task if possible
- [ ] unit tests

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/tasks/tasks.ts [hunk 1]
```diff
@@ -21,13 +32,19 @@ export class Tasks extends Component {
...
+  private readonly makefile?: Makefile;
+
+  constructor(project: Project, options: TasksOptions = {}) {
     super(project);
 
     const manifestFile = Tasks.MANIFEST_FILE;
     this._tasks = {};
     this._env = {};
 
+    if (options.makefile) {
```
Comment (eladb) [inline src/tasks/tasks.ts:44]:
```suggestion
    if (options.makefile ?? false) {
```

File: src/tasks/tasks.ts [hunk 2]
```diff
@@ -115,4 +132,85 @@ export class Tasks extends Component {
...
+   */
+  public getFullEnvironment(task: Task) {
+    let env = this._env ?? {};
+
+    env = {
+      ...env,
+      ...task._renderSpec().env ?? {},
+    };
+
+    return env;
```
Comment (eladb) [inline src/tasks/tasks.ts:148]:
```suggestion
    return {
      ...this._env,
      ...task._renderSpec().env,
    };
```

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,145 @@
+.PHONY: all
```
Comment (eladb) [inline Makefile:1]: Projen marker

**Review by eladb — COMMENTED** [review]
I like this direction but I don't like that we need to maintain two runtimes. I am wondering if we should only use make files to run tasks. Could be a better long term approach...

**Comment by Chriscbr** [comment]
Time permitting I'll try to return to the PR this weekend

**Comment by Chriscbr** [comment]
I'm planning to revisit this soon, so keep your eyes peeled

**Pull request closed without merging**